cloudfour/cloudfour.com-patterns
# Basic Storybook setup

## Overview

This populates the new branch with a basic Storybook setup. It includes two demo components, one containing Twig and SCSS files, the other using a basic JS setup returning HTML. The following non-standard features have been configured:

- Doc pages via MDX
- Accessibility tests per story
- Configurable "knobs" for testing parameters
- Viewport resizing per story
- SCSS loading using flagship Sass (instead of the default, slightly slower version)
- Streamlined Twig file support via `twigjs-loader`

I'm sure we'll want to continue tweaking these things as we move forward, but I think this gets us what we need to start with.

## Known issues

- Documentation previews are stupid tall. They are implementing automatic resizing for plain HTML in a future update, and there are short-term fixes, but I didn't want to get hung up on that now.
- This branch will probably _not_ pass Netlify status checks and deploy previews because Netlify has no way of knowing the new build command, output file location, etc.
- Storybook by default installs everything with `--save-dev`. That _might_ or _might not_ make sense depending on which dependencies are also required for consumable output. I just followed their lead for now.
- No custom UI or logo or anything yet! I figure once we have those assets as patterns in the project, that'll be faster.
- The viewport add-on is using the default options. These can be configured later to be relevant to our project.
- There is no concept right now of "base" styles or shared things between components. More work may be required to get that working.
- There may be opportunities to specify default decorators or settings across files or to generate stories automatically, but I didn't want to get too deep into that quite yet.

## Testing

1. Check out branch locally.
1. `npm ci`
1. `npm run storybook`
1. Observe Storybook in browser.
1. Try editing some files, make sure it generally works.

---

Resolves #467
